### DISCUSSION TITLE
Debug type .dll and release type .dll have the same name in some package.

### DISCUSSION THREAD

**Discussion by HWZen** [microsoft/vcpkg#26002] [Q&A] [answered]
There is a problem that has been bothering me for a long time.

I install some package via vcpkg, such as gtest, and its debug type .dll (in ```vcpkg\installed\x64-windows\debug\bin```) and release type .dll (in ```vcpkg\installed\x64-windows\bin```) have the same name , and I'm used to add all bin path to system variables ”path“.

If ```vcpkg\installed\x64-windows\debug\bin``` is precedes ```vcpkg\installed\x64-windows\bin```，test throw exception in release mode, vice versa.

Every time I switch test mode I have to switch the path order.
Sometimes I really find it tedious.

Is it a good idea to copy the dll to each of my projects?

So does anyone have a good idea to help me with this?

**Comment by JackBoosY** [comment]
Yeah it's a common issue here. Maybe the best way is make a way to provide the specific library which contains the absolute library path.

**Comment by dg0yt** [comment, marked as answer]
I try hard to have `PATH` not include every possible unsafe place...
vcpkg has features to deploy the runtime next to your executables, at least as part of its cmake toolchain.

**Reply by HWZen** [reply]
Oh yes !
I don't need to add  *.dll dir to PATH.
cmake (or vcpkg) will copy them to executor dir.

And in Linux, vcpkg build gtest to a static lib.

emmm~

Anyway, my problem has been solved
